core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -1343,6 +1350,14 @@ extern "C" fn new_show_busyscreen(n_args: usize, args: *const Obj, kwargs: *mut
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_show_welcome_screen(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
```

Comment:
Rename to `draw_welcome_screen` and have it call `paint()` directly. We don't even need to instantiate a `LayoutObj` for it: just create the `WelcomeScreen` on stack, call `frame.place(consts::screen())` and `frame.paint()` and return.
(see also [this](https://github.com/trezor/trezor-firmware/blob/c995a9a359f57f9e7cdc19142e73380907945059/core/embed/rust/src/ui/model_tt/bootloader/mod.rs#L102-L115))

Reply:
(that way we will eventually be able to move the draw call to C without doing anything fancy)

it also prooobably doesn't need the `try` thing? if you don't try to allocate, there is nothing that could fail

Reply:
Both done in [5601deb](https://github.com/trezor/trezor-firmware/pull/2840/commits/5601debbbc46216c04794814e5199550cfdf9bca)